### DISCUSSION TITLE
How to use faker with Node<=12 and Typescript without deprecated @types/faker?

### DISCUSSION THREAD

**Discussion by migsoares** [faker-js/faker#924] [Q&A] [answered]
Due to the recent changes in in this project I needed to update to 'v5.5.3` since it's the last one that supports Node<=12 (Node<=12 is a requirement for my project).

Followed the instructions for the [TypeScript Support](TypeScript Support) for versions prior to v6 but I started to get the error `Cannot find module 'faker' or its corresponding type declarations.` even though I have added `@types/faker`.

To fix this realized that I need to use [`@types/faker@5.5.9`](https://cdn.jsdelivr.net/npm/@types/faker@5.5.9/) because it's the last one that has the `index.d.ts`.

Since [`@types/faker`](https://yarnpkg.com/package/@types/faker) is deprecated I would like to know if there is an alternative to use `@faker-js/faker` with Node<=12 without the need of `@types/faker` nor `faker.d.ts`.

Thanks 🙏🏻

**Comment by Shinigami92** [comment, marked as answer]
No there is no alternative
If you need to support legacy software, you need to use legacy dependencies 🤷 

There will no support anymore by us for Node < v14.6 and TS < 4.0
